assignment4/assignment4.cpp

```diff
@@ -0,0 +1,54 @@
+#include <iostream>
+#include <math.h>
+#include <vector>
+
+class IslandCounter{
+	void collapse_island(std::vector< std::vector<bool> > &map, int i, int j){
+		if(!map[i][j]) return;
+
+		map[i][j] = false;
```

Comment:
It may be a bit more clear (or not :) ) to do a loop here over the 4 cardinal directions (either using two arrays for delta_x & delta_y, or looping over all 8 directions and only using the ones you need), and check if it's out of bonds at the beginning of the function.